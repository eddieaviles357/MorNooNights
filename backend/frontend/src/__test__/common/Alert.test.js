import { render, cleanup } from '@testing-library/react';
import Alert from '../../components/common/Alert';

afterEach(cleanup);

describe("Alert component", () => {
  it('renders without crashing', () => {
    render(<Alert />);
  });
  
  it('matches snapshot', () => {
    const { asFragment } = render(<Alert />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('matches snapshot for error message', () => {
    const { asFragment } = render(<Alert messages={["error"]}/>);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it("matches snapshot for success mesage", function() {
    let messages = ["Green"];
    const { asFragment } = render(<Alert type="success" messages={messages} />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it("messages are displayed", function() {
    let messages = ["error"];
    const alert = render(<Alert type="success" messages={messages} />);
    expect(alert.queryByText('error')).toBeInTheDocument();
  });
})
