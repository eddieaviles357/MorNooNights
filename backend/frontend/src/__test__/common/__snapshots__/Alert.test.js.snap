// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  />
</DocumentFragment>
`;

exports[`Alert component matches snapshot for error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      error
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert component matches snapshot for success mesage 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      Green
    </p>
  </div>
</DocumentFragment>
`;
